src/ImageManipulation.php

```diff
@@ -1057,4 +1062,98 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    private function copyImageBackend()
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $file->setImageBackend($this->getImageBackend());
+        return $file->setOriginal($this);
+    }
+
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->setMutableAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    public function setMutableAttributes($attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    protected function getDefaultAttributes(): array
+    {
+        if (!$this->getIsImage()) {
+            return [];
+        }
+
+        $attributes = [
+            'width' => $this->getWidth(),
+            'height' => $this->getHeight()
+        ];
+
+        if ($this->getIsLazyLoaded()) {
+            $attributes['loading'] = 'lazy';
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * Return whether image will be lazy loaded
+     *
+     * @return bool
+     */
+    public function getIsLazyLoaded() : bool
+    {
+        if (Image::getLazyLoadingEnabled() && $this->getIsImage()) {
+            return empty($this->attributes['loading']) || $this->attributes['loading'] === 'lazy';
+        }
+        return false;
+    }
+
+    public function LazyLoad($lazyLoad): AssetContainer
+    {
+        if (!Image::getLazyLoadingEnabled() || !$this->getIsImage()) {
+            return $this;
+        }
+
+        if (is_string($lazyLoad)) {
+            $lazyLoad = strtolower($lazyLoad);
+        }
+
+        switch ($lazyLoad) {
+            case false:
+            case 0:
+            case '0':
+            case 'false':
+            case 'eager':
+                return $this->setAttribute('loading', 'eager');
+
+            case true:
```

Comment:
+1 for adding the truthy conditions here

just double checking - is a switch statement a strict === check?

Reply:
Good question, I'll have a look.

Reply:
`switch` does a loose comparison so I had to scrap this approach.